### PR TITLE
Update docs for Spark 3/Scala 2.12

### PR DESCRIPTION
**Pull request by karenfeng**
## What changes are proposed in this pull request?

Small docs changes to reflect that our default build is on Spark 3/Scala 2.12, and to suggest that users build on Spark 3. Should merge only after the next Spark 3-based release.

## How is this patch tested?
- [ ] Unit tests
- [ ] Integration tests
- [x] Manual tests

Manually installed Spark 3.0.1 and confirmed the quickstart guide works.

### CODE REVIEW

File: docs/source/getting-started.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ Glow requires Apache Spark 2.4.3 (or a later version of Spark 2.4.x that is buil
 
         .. code-block:: sh
 
-          pip install pyspark==2.4.3
+          pip install pyspark==3.0.0
```
Comment (henrydavidge) [inline docs/source/getting-started.rst:17]: This doesn't work until we publish a 3.0 artifact, does it?

Comment (karenfeng) [reply docs/source/getting-started.rst:17]: Yes - I'll clarify that this is blocked on the next release.

**Review by kianfar77 — APPROVED** [review]
LGTM

**Review by henrydavidge — APPROVED** [review]
Thanks a lot!

**Comment by karenfeng** [comment]
Merging as the 0.6.0 release is public.

**Pull request merged**